test/NuGet.Clients.Tests/NuGet.VisualStudio.Common.Test/Telemetry/TransitiveDependenciesCounterfactualEvent.cs

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Moq;
+using NuGet.Common;
+using Xunit;
+
+namespace NuGet.VisualStudio.Common.Test
+{
+    public class TransitiveDependenciesCounterfactualEventTests
+    {
+        [Fact]
+        public void EmitEvent_Succeeds()
```

Comment:
nit: I don't get how this test adds much :D

Reply:
It's a smoke test.

Reply:
I see there is an existing pattern to test telemetry events like this but looking more closely, the EmitTelemetryEvent method just null checks its parameters then delegates to _telemetrySession.PostEvent which is being mocked in these tests so there's not much product code being tested. The lastTelemetryEvent is a test variable so testing that what we supplied the mock service is the same thing as the mock service is returning to us is testing test code more than product code.

Reply:
It looks like we have a pattern of testing telemetry events by making sure the event we send the telemetry service, which is mocked in these tests, sends the event to the telemetry session. This could be covered in a single test on the telemetry session and then every single event added wouldn't need to test the telemetry session is sending the event it was given in the PostEvent method.

Reply:
Removed. Replaced with existing test to check that event is really emitted.